### ISSUE TITLE
Character encoding error when exec'ing a tar.gz && control code is missing UTF-8 declaration

### ISSUE THREAD

**Issue by james-stocks** [inspec/inspec#2938]
### 🎛 Description

Given the following control code that contains a `´` character:

```
# copyright: 2017, The Authors

title 'sample section'

# you can also use plain tests
describe file('/tmp/foo') do
  skip 'There is no cow file inside sam´s home directory.'
end
```

The control works when executed from source but throws an error when executed from .tar.gz:

```
➜  /tmp > inspec exec test_profile

Profile: InSpec Profile (test_profile)
Version: 0.1.0
Target:  local://

  File /tmp/foo
     ↺  There is no cow file inside sam´s home directory.

Test Summary: 0 successful, 0 failures, 1 skipped
➜  /tmp > inspec archive test_profile
I, [2018-04-10T15:20:15.057507 #52365]  INFO -- : Checking profile in test_profile
I, [2018-04-10T15:20:15.057675 #52365]  INFO -- : Metadata OK.
I, [2018-04-10T15:20:15.058602 #52365]  INFO -- : Found 1 controls.
I, [2018-04-10T15:20:15.058631 #52365]  INFO -- : Control definitions OK.
I, [2018-04-10T15:20:15.058740 #52365]  INFO -- : Generate archive /private/tmp/test_profile-0.1.0.tar.gz.
I, [2018-04-10T15:20:15.060838 #52365]  INFO -- : Finished archive generation.
➜  /tmp > inspec exec test_profile-0.1.0.tar.gz
/opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:144:in `format_message': incompatible character encodings: UTF-8 and ASCII-8BIT (Encoding::CompatibilityError)
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:128:in `format_result'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:91:in `block (2 levels) in print_anonymous_control_results'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:90:in `each'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:90:in `block in print_anonymous_control_results'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:86:in `each'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:86:in `print_anonymous_control_results'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:51:in `block in render'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:46:in `each'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters/cli.rb:46:in `render'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/reporters.rb:30:in `render'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/runner.rb:111:in `block in render_output'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/runner.rb:110:in `each'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/runner.rb:110:in `render_output'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/runner.rb:134:in `run_tests'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/runner.rb:104:in `run'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/lib/inspec/cli.rb:168:in `exec'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /opt/inspec/embedded/lib/ruby/gems/2.4.0/gems/inspec-2.1.30/bin/inspec:12:in `<top (required)>'
	from /usr/local/bin/inspec:72:in `load'
	from /usr/local/bin/inspec:72:in `<main>'
➜  /tmp >
```

The error does not occur if `# encoding: UTF-8` is placed at the top of the control file.

### 🌍 InSpec and Platform Version

2.1.30
Testing directly against my OSX laptop, but issue was first seen on a RHEL7 host.

**Comment by jquick** [comment]
Thanks for reporting this @james-stocks

**Comment by james-stocks** [comment]
https://github.com/inspec/inspec/issues/1659 would make the change so we always extract tar.gz/zips

**Comment by chris-rock** [comment]
fixed by #3376
